ov/appliance_device_snmp_v3_user.go

```diff
@@ -0,0 +1,240 @@
+		return snmpv3User, err
+	}
+
+	if err != nil {
+		return snmpv3User, err
+	}
+
+	return snmpv3User, nil
+}
+
+func (c *OVClient) GetSNMPv3Users(start string, count string, filter string, sort string) (SNMPv3UserList, error) {
+	var (
+		uri            = "/rest/appliance/snmpv3-trap-forwarding/users"
+		q              = make(map[string]interface{})
+		snmpv3userlist SNMPv3UserList
+	)
+
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	if len(filter) > 0 {
+		q["filter"] = filter
+	}
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	if start != "" {
+		q["start"] = start
+	}
+
+	if count != "" {
+		q["count"] = count
+	}
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return snmpv3userlist, err
+	}
+
+	log.Debugf("Get v1 Trap list: %s", data)
+	if err := json.Unmarshal(data, &snmpv3userlist); err != nil {
+		return snmpv3userlist, err
+	}
+	return snmpv3userlist, nil
+
+}
+
+func (c *OVClient) GetSNMPv3UserById(id string) (SNMPv3User, error) {
+	var (
+		uri          = "/rest/appliance/snmpv3-trap-forwarding/users/" + id
+		snmpv3userid SNMPv3User
+	)
+
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return snmpv3userid, err
+	}
+
+	log.Debugf("Get SNMP users By Id %s", data)
+	if err := json.Unmarshal(data, &snmpv3userid); err != nil {
+		return snmpv3userid, err
+	}
+	return snmpv3userid, nil
+}
+
+func (c *OVClient) GetSNMPv3UserByUserName(username string) (SNMPv3User, error) {
+	var (
+		snmpv3userusername SNMPv3User
+	)
+
+	snmpv3userusernames, err := c.GetSNMPv3Users("", "", fmt.Sprintf("userName matches '%s'", username), "userName:asc")
+	if snmpv3userusernames.Total > 0 {
+		return snmpv3userusernames.Members[0], err
+	} else {
+		return snmpv3userusername, err
+	}
+
+}
+
+func (c *OVClient) UpdateSNMPv3User(updateOption SNMPv3User, id string) (SNMPv3User, error) {
+	var (
+		uri            = "/rest/appliance/snmpv3-trap-forwarding/users/" + id
+		updateResponse SNMPv3User
+	)
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	log.Debugf("REST : %s \n %+v\n", uri, updateOption)
+	data, err := c.RestAPICall(rest.PUT, uri, updateOption)
+	if err != nil {
+		log.Errorf("Error submitting update SNMPv3 User  request: %s", err)
+		return updateResponse, err
+	}
+
+	log.Debugf("Response SNMPv3 User %s", data)
+
+	if err := json.Unmarshal(data, &updateResponse); err != nil {
+		return updateResponse, err
+	}
+
+	return updateResponse, nil
+}
+
+func (c *OVClient) DeleteSNMPv3UserById(id string) error {
+	var (
+		snmpv3User SNMPv3User
+		err        error
+		uri        string
+	)
+
+	snmpv3User, err = c.GetSNMPv3UserById(id)
+
+	if err != nil {
+		return err
+	}
+	if snmpv3User.URI != "" {
+		log.Debugf("REST : %s \n %+v\n", snmpv3User.URI, snmpv3User)
+
+		uri = snmpv3User.URI.String()
+		if uri == "" {
```

Comment:
Whats the use of condition since we are snmpv3User.URI != "" above?

Reply:
We are connverting the URI to string so, another check is used.